packages/costa/src/runtime.ts

```diff
@@ -42,11 +42,35 @@ function selectNpmPackage(metadata: NpmPackageMetadata, source: PluginSource): N
   return metadata.versions[metadata['dist-tags'].latest];
 }
 
-function spawnAsync(command: string, args?: string[], opts: SpawnOptions = {}): Promise<string> {
+interface LogWriter {
+  stdout: Writable;
+  stderr: Writable;
+}
+
+interface CostaSpawnOptions extends SpawnOptions {
+  logWriter: LogWriter;
+}
+
+function pair(read: Readable, write: Writable) {
+  read.on('error', (err) => {
```

Comment:
I think we need to document why do we use this style of the block instead of using `pipe()`.